### PR TITLE
rpc: fix 0-ady functions raising an exception

### LINKED ISSUE #130: rpccall with 0-ady functions fails
# Description

`@rpccall` for functions that take no arguments throws an error when defining a function with no arguments. 

# To reproduce

Try to run the following file:

```
from p4p.rpc import rpcproxy, rpccall

@rpcproxy
class MyProxy(object):
    @rpccall("%s:foo")
    def foo():
        pass
```

# Observed behaviour

This throws the following error:

```
$ python3 /tmp/p4p_rpcproxy_noargs.py 
Traceback (most recent call last):
  File "/tmp/p4p_rpcproxy_noargs.py", line 4, in <module>
    class MyProxy(object):
  File "/home/training/dev/epics-rpc-ads-proxy/p4ptest/lib/python3.8/site-packages/p4p/rpc.py", line 317, in rpcproxy
    obj[K] = _wrapMethod(K, V)
  File "/home/training/dev/epics-rpc-ads-proxy/p4ptest/lib/python3.8/site-packages/p4p/rpc.py", line 267, in _wrapMethod
    if len(S.args) != len(S.defaults):
TypeError: object of type 'NoneType' has no len()
```

# Expected behaviour

Runs without error, allowing the user to access the function over PVA.

### PR DESCRIPTION
**Pull request by Synthetica9**
Closes #130

### CODE REVIEW

File: src/p4p/rpc.py [hunk 1]
```diff
@@ -264,13 +264,14 @@ def _wrapMethod(K, V):
     if S.varargs is not None or keywords is not None:
         raise TypeError("vararg not supported for proxy method %s" % K)
 
-    if len(S.args) != len(S.defaults):
+    args, defaults = ([] if v is None else v for v in (S.args, S.defaults))
```
Comment (mdavidsaver) [inline src/p4p/rpc.py:267]: Do you mean for this line to do something like the following?

```py
args, defaults = args or [], defaults or []
```

**Comment by mdavidsaver** [comment]
For this PR and #131, it would be helpful to have added test coverage to `TestProxy` in `src/p4p/test/test_rpc.py`.  Including cases where eg. the client-side proxy should throw an exception due to malformed call arguments.

File: src/p4p/rpc.py (hunk 1 above)
Comment (Synthetica9) [reply src/p4p/rpc.py:267]: Yes, but I generally don't use the `or` syntax to handle `None` because it also brakes on falsely values.

Comment (mdavidsaver) [reply src/p4p/rpc.py:267]: For reference, I understand the concern about loose type comparison rules.  In this instance, `S` comes from the builtin `inspect` module, which I think can be relied on not to change radically.  So I think the `or []` is a reasonable abbreviation.

**Comment by mdavidsaver** [comment]
Thank you for this PR.  However, I have applied a different fix with 4f6b9e9e67293acd142cf1ce377c87da9e6ab549.

**Pull request closed without merging**